src/rust/engine/src/nodes.rs

```diff
@@ -1288,8 +1288,11 @@ impl NodeError for Failure {
     Failure::Invalidated
   }
 
-  fn cyclic() -> Failure {
-    throw("Dep graph contained a cycle.")
+  fn cyclic(path: Vec<String>) -> Failure {
+    throw(&format!(
+      "Dep graph contained a cycle:\n  {}",
+      path.join("\n  ")
```

Comment:
Given
```
+target(
+name = 'a',
+dependencies = [':b'])
+
+target(
+name = 'b',
+dependencies = [':c'])
+
+target(
+name = 'c',
+dependencies = [':a'])
```
Optional UX nit for the following output:
```
          Task(transitive_hydrated_target(), //:c, TransitiveHydratedTarget, true)
          Task(transitive_hydrated_target(), //:b, TransitiveHydratedTarget, true)
          Task(transitive_hydrated_target(), //:a, TransitiveHydratedTarget, true)
```
if we could single out the target addresses and show the directions, that may be more clear to users. E.g.
```
//:a -> //:b -> //:c -> //:a
```

Reply:
+1

Reply:
It would be, yea. But I think that's squarely in the camp of #7907. Nodes in general do not have better string representations right now.

Reply:
sgtm. thanks!